api/src/app/modules/submission/submission.service.ts

```diff
@@ -668,13 +668,39 @@ export class SubmissionService extends DataService<Submission, Repository<Submis
     }
   }
 
+  private async simplifyGeometry(submissionID: number, spatialObjectCode: SpatialObjectCodeEnum) {
+    this.logger.debug(`Simplify geometry for submittion: ${submissionID}, spatial object type: ${spatialObjectCode}`);
+    try {
+      await this.getDataSource()
+        .query(
+          `
+            UPDATE app_fom.${SpatialObjectTableEnum[spatialObjectCode]} SET geometry=ST_SimplifyPreserveTopology(geometry, 2.5) where submission_id = ${submissionID};
+          `,
+        );
+      this.logger.debug(`Simplify geometry successfully`);
+    }
+    catch (error) {
+      throw new BadRequestException(`Failed on simplifying geometry for submittion: ${submissionID}, spatial object type: ${spatialObjectCode}: ${error}`);
+    }
+  }
+
   private async saveAndUpdateSpatialSubmission(
     updatedSubmission: Submission, 
     spatialObjectCode: SpatialObjectCodeEnum, 
-    user: User) {
+    user: User
+  ) {
     await this.repository.save(updatedSubmission);
     await this.updateProjectLocation(updatedSubmission.projectId, user);
     await this.updateGeospatialAreaOrLength(spatialObjectCode, updatedSubmission.id);
   }
 
+  private async saveAndSimplifySpatialSubmission (
+    submission: Submission, 
+    spatialObjectCode: SpatialObjectCodeEnum, 
+    user: User
+  ) {
+    await this.saveAndUpdateSpatialSubmission(submission, spatialObjectCode, user);
+    await this.simplifyGeometry(submission.id, spatialObjectCode);
```

Comment:
The current logic does the simplification after calculating the area & length, I'd prefer to simplify before.